src/DmaInputFilter.cc

```diff
@@ -0,0 +1,132 @@
+#include <cassert>
+#include <cstdio>
+#include <cerrno>
+#include <system_error>
+#include <iostream>
+#include <sstream>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "DmaInputFilter.h"
+#include "log.h"
+
+
+
+DmaInputFilter::DmaInputFilter( const std::string& deviceFileName, size_t packetBufferSize, size_t nbPacketBuffers, ctrl& control ) : 
+  InputFilter( packetBufferSize, nbPacketBuffers, control )
+{ 
+  dma_fd = open( deviceFileName.c_str(), O_RDWR | O_NONBLOCK );
+  if ( dma_fd < 0 ) {
+    throw std::system_error(errno, std::system_category(), "Cannot open DMA device: " + deviceFileName);
+  }
+
+  LOG(TRACE) << "Created DMA input filter"; 
+}
+
+DmaInputFilter::~DmaInputFilter() {
+  close( dma_fd );
+  LOG(TRACE) << "Destroyed DMA input filter";
+}
+
+/*
+ * man 2 write:
+ * On Linux, write() (and similar system calls) will transfer at most
+ * 	0x7ffff000 (2,147,479,552) bytes, returning the number of bytes
+ *	actually transferred.  (This is true on both 32-bit and 64-bit
+ *	systems.)
+ */
+
+#define RW_MAX_SIZE	0x7ffff000
+
+static inline ssize_t read_axi_packet_to_buffer(int fd, char *buffer, uint64_t size)
+{
+  ssize_t rc;
+  uint64_t to_read = size;
+
+  if (to_read > RW_MAX_SIZE) {
+    to_read = RW_MAX_SIZE;
+  }
+
+  /* read data from file into memory buffer */
+  rc = read(fd, buffer, to_read);
+
+  if (rc <= 0) {
+    return rc;
+  }	
+  return rc;
+}
+
+ssize_t DmaInputFilter::readPacketFromDMA(char **buffer, size_t bufferSize)
+{
+  // Read from DMA
+  // We need at least 1MB buffer
```

Comment:
I *think* this comment is left over from the previous version. Do we still check somewhere that we have a sufficient buffer size?

Reply:
Ah, found the check. Still, the comment should be removed.

Reply:
Check for the correct buffer size moved to the main function